src/test/java/com/sendgrid/SendGridTest.java

```diff
@@ -3242,4 +3242,50 @@ public void test_request_headers_override_sendgrid_object_headers() throws IOExc
     sg.api(request);
     verify(client).api(argThat((Request req) -> req.getHeaders().get("set-on-both").equals("456")));
   }
+
+  @Test
+  public void testSetResidency_happy_path_eu() {
+    SendGrid sg = new SendGrid(SENDGRID_API_KEY);
+    sg.setDataResidency("eu");
+    Assert.assertEquals(sg.getHost(), "api.eu.sendgrid.com");
+  }
+  @Test
+  public void testSetResidency_happy_path_global() {
+    SendGrid sg = new SendGrid(SENDGRID_API_KEY);
+    sg.setDataResidency("global");
+    Assert.assertEquals(sg.getHost(), "api.eu.sendgrid.com");
```

Comment:
commented in the other PR as well, shouldn't this be `api.sendgrid.com`?

Reply:
Yes, correct. Updated this testcase.